manager/vm/vm.go

```diff
@@ -17,10 +18,40 @@ type VM interface {
 	SetProcess(pid int) error
 	GetProcess() int
 	GetCID() int
-	Transition(newState manager.ManagerState) error
+	Transition(newState pkgmanager.ManagerState) error
 	State() string
 	GetConfig() interface{}
 }
 
 //go:generate mockery --name Provider --output=./mocks --filename provider.go --quiet --note "Copyright (c) Ultraviolet \n // SPDX-License-Identifier: Apache-2.0"
-type Provider func(config interface{}, logsChan chan *manager.ClientStreamMessage, computationId string) VM
+type Provider func(config interface{}, eventSender EventSender, computationId string) VM
+
+type EventsLogs interface {
+	IsEventLog() bool
+}
```

Comment:
This interface seems weird. What's the purpose of this method except to provide a wrapper around events and logs? Do events and logs share any common behavioral pattern (say, can be seralized)?

Reply:
Yes they can be serilized, but there is is not higher purpose to this interface, maybe removing the type here is the way to go so its just interface{}